content/frontend/react-urql/1-getting-started.md

```diff
@@ -0,0 +1,554 @@
+
+</Instruction>
+
+```js{4}(path=".../hackernews-react-urql/src/index.js")
+import React from 'react'
+import ReactDOM from 'react-dom'
+import './styles/index.css'
+import App from './components/App'
+```
+
+```js{2}(path=".../hackernews-react-urql/src/components/App.js")
+import React, { Component } from 'react';
+import logo from '../logo.svg';
+import '../styles/App.css';
+```
+
+Your project structure should now look as follows:
+
+```bash(nocopy)
+.
+├── README.md
+├── node_modules
+├── package.json
+├── public
+│   ├── favicon.ico
+│   ├── index.html
+│   └── manifest.json
+├── src
+│   ├── App.test.js
+│   ├── components
+│   │   └── App.js
+│   ├── index.js
+│   ├── logo.svg
+│   ├── serviceWorker.js
+│   └── styles
+│       ├── App.css
+│       └── index.css
+└── yarn.lock
+```
+
+#### Prepare styling
+
+This tutorial is about the concepts of GraphQL and how you can use it from within a React application, so we want to spend the least time possible on styling. To reduce the usage of CSS in this project, you'll use the [Tachyons](http://tachyons.io/) library which provides a number of CSS classes.
+
+<Instruction>
+
+Open `public/index.html` and add a third `link` tag right below the two existing ones that pulls in Tachyons:
+
+```html{3}(path=".../hackernews-react-urql/public/index.html")
+<link rel="manifest" href="%PUBLIC_URL%/manifest.json">
+<link rel="shortcut icon" href="%PUBLIC_URL%/favicon.ico">
+<link rel="stylesheet" href="https://unpkg.com/tachyons@4.2.1/css/tachyons.min.css"/>
+```
+
+</Instruction>
+
+Since we still want to have a bit more custom styling here and there, we also prepared some styles for you that you need to include in the project.
+
+<Instruction>
+
+Open `index.css` and replace its content with the following:
+
+```css(path=".../hackernews-react-urql/src/styles/index.css")
+body {
+  margin: 0;
+  padding: 0;
+  font-family: Verdana, Geneva, sans-serif;
+}
+
+input {
+  max-width: 500px;
+}
+
+.gray {
+  color: #828282;
+}
+
+.orange {
+  background-color: #ff6600;
+}
+
+.background-gray {
+  background-color: rgb(246,246,239);
+}
+
+.f11 {
+  font-size: 11px;
+}
+
+.w85 {
+  width: 85%;
+}
+
+.button {
+  font-family: monospace;
+  font-size: 10pt;
+  color: black;
+  background-color: buttonface;
+  text-align: center;
+  padding: 2px 6px 3px;
+  border-width: 2px;
+  border-style: outset;
+  border-color: buttonface;
+  cursor: pointer;
+  max-width: 250px;
+}
+```
+
+</Instruction>
+
+#### Install urql
+
+<Instruction>
+
+Next, you need to pull in the functionality of `urql`. We'll also be installing
+two "Exchanges" for urql, which we'll use to set up normalized caching and support
+for React Suspense:
+
+```bash(path=".../hackernews-react-urql")
+yarn add urql @urql/exchange-graphcache @urql/exchange-suspense graphql
```

Comment:
Maybe we need to add graphql-tag here as well. I used it in the final solution, we could decide to remove it

Reply:
Yep, you're right! That's totally missing